shared/actions/signup.js

```diff
@@ -1,366 +1,198 @@
   }
-}
 
-function signup(skipMail: boolean, onDisplayPaperKey?: () => void) {
-  return (dispatch, getState) => {
-    const p = new Promise((resolve, reject) => {
-      const {email, username, inviteCode, passphrase, deviceName} = getState().signup
-      paperKeyResponse = null
-      const deviceType = isMobile ? RPCTypes.commonDeviceType.mobile : RPCTypes.commonDeviceType.desktop
-
-      if (email && username && inviteCode && passphrase && deviceName) {
-        RPCTypes.signupSignupRpcPromise({
-          incomingCallMap: {
-            'keybase.1.gpgUi.wantToAddGPGKey': (params, response) => {
-              // Do not add a gpg key for now
-              response.result(false)
-            },
-            'keybase.1.loginUi.displayPrimaryPaperKey': ({sessionID, phrase}, response) => {
-              paperKeyResponse = response
-              dispatch(SignupGen.createShowPaperKey({paperkey: new HiddenString(phrase)}))
-              onDisplayPaperKey && onDisplayPaperKey()
-              dispatch(nextPhase())
-            },
+  return Saga.call(function*() {
+    try {
+      yield RPCTypes.signupSignupRpcSaga({
+        incomingCallMap: {
+          // Do not add a gpg key for now
+          'keybase.1.gpgUi.wantToAddGPGKey': (params, response, state) => {
+            response.result(false)
           },
-          deviceName,
-          deviceType,
+          // We dont show the paperkey anymore
+          'keybase.1.loginUi.displayPrimaryPaperKey': () => {},
+        },
+        params: {
+          deviceName: devicename,
+          deviceType: isMobile ? RPCTypes.commonDeviceType.mobile : RPCTypes.commonDeviceType.desktop,
           email,
           genPGPBatch: false,
           genPaper: false,
           inviteCode,
           passphrase: passphrase.stringValue(),
-          skipMail,
+          skipMail: false,
           storeSecret: true,
           username,
-          waitingHandler: isWaiting => {
-            dispatch(SignupGen.createWaiting({waiting: isWaiting}))
-          },
-        })
-          .then(({passphraseOk, postOk, writeOk}) => {
-            logger.info('Successful signup', passphraseOk, postOk, writeOk)
-            dispatch(SignupGen.createWaiting({waiting: true}))
-            resolve()
-          })
-          .catch(err => {
-            logger.warn('error in signup:', err)
-            dispatch(SignupGen.createSignupError({signupError: new HiddenString(err.desc)}))
-            dispatch(nextPhase())
-            reject(new Error(err))
-          })
-      } else {
-        logger.warn('Entered signup action with a null required field')
-        reject(new Error('null required field'))
-      }
-    })
-    return p
-  }
+        },
+        waitingKey: Constants.waitingKey,
+      })
+      yield Saga.put(SignupGen.createSignedup())
+    } catch (error) {
+      yield Saga.put(
+        SignupGen.createSignedupError({error: new HiddenString(`Cant signup, try again? ${error.desc}`)})
```

Comment:
The page already renders `Ah Shoot! something went wrong...`, so the `Cant signup, try again?` is redundant here.
<img width="532" alt="image" src="https://user-images.githubusercontent.com/11968340/42169158-ee462d6e-7de0-11e8-8b22-e77dd25615e8.png">